### PR TITLE
Preserve system annotations and labels during object update

### LINKED ISSUE #564: Unreachable code in create stage provider
https://github.com/eclipse-symphony/symphony/blob/ccf7804b126ae531a3c6e5afb686743b0b2e004d/api/pkg/apis/v1alpha1/providers/stage/create/create.go#L327C3-L327C4

### PR DESCRIPTION
**Pull request by linyguo**
Fix https://github.com/eclipse-symphony/symphony/issues/561
Fix #564 together

### CODE REVIEW

File: api/constants/constants.go
Comment (Haishi2016) [inline api/constants/constants.go:32]: If this deem to be a system annotation, should it be put under management.azure.com/ as well?

File: api/pkg/apis/v1alpha1/providers/stage/create/create.go
Comment (linyguo) [reply api/pkg/apis/v1alpha1/providers/stage/create/create.go:326]: we have the same log in line 285, so it is safe to delete it here

File: api/constants/constants.go
Comment (linyguo) [reply api/constants/constants.go:24]: updated k8s related consts to reference api constants.

File: api/pkg/apis/v1alpha1/managers/activations/activations-manager.go
Comment (RemindD) [inline api/pkg/apis/v1alpha1/managers/activations/activations-manager.go:158]: Why don't you use "GetState" interface? In addition, there is another call in the ValidateCreateOrUpdate to get previous state which seems duplicate.

Comment (linyguo) [reply api/pkg/apis/v1alpha1/managers/activations/activations-manager.go]: updated.

**Review by msftcoderdjw — APPROVED** [review]

**Review by msftcoderdjw — COMMENTED** [review]
Discussed with @RemindD before merging, still have some comments. @RemindD  please post your comment here.

File: api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go
Comment (RemindD) [inline api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go:268]: Discussed with Jiawei offline, we think we need to use entry.Value.Etag in resourceVersion even when it's empty

File: api/constants/constants.go
Comment (Haishi2016) [inline api/constants/constants.go:38]: I'm concerned about Azure specific constants/logics get leaked into OSS. We had this azure feature flag that is designed to be used to compile Azure specific version. Now this is mingled with OSS.

Comment (msftcoderdjw) [reply api/constants/constants.go:38]: I agree with this. But I don't want this to block this PR because this PR also preserve the OSS system managed annotation. @linyguo will open another issue, let's have discussion when handling that issue.

Comment (linyguo) [reply api/constants/constants.go:38]: Created #584 to follow up this issue

File: api/pkg/apis/v1alpha1/providers/states/k8s/k8s_test.go
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/providers/states/k8s/k8s_test.go:105]: Per offline discussion, there's no need to pass etag in creation case. Let's skip it.

**Review by msftcoderdjw — CHANGES_REQUESTED** [review]

**Review by RemindD — APPROVED** [review]

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**
